GIT & CLI Commands (#9)
carolinacodeschool/foundations-front-end-engineering: coursework/1.0 Day One - Setup Instructions and an Introduction to HTML and CSS.md

* **<a href="https://code.visualstudio.com/download">Visual Studio Code (VS Code) (a free, open-source code editor built by Microsoft</a>**
<!-- GitHub Desktop - a well-designed application for managing projects that are version-controlled with Git -->

<!-- You'll also need to sign up for a free GitHub account. Signing up for GitHub and installing these two tools should only take a few minutes, but they'll make a huge difference in your success. -->

## GIT AND GITHUB

>  GitHub is popular web-based hosting service for version control using Git, offers additional features such as pull requests, issue tracking, and project management tools, facilitating seamless collaboration and efficient software development workflows.

Reviewer: ```suggestion
 GitHub is a popular web-based hosting service for version control which offers features such as pull requests, issue tracking, and project management tools, facilitating seamless collaboration and efficient software development workflows. You'll need to create an account to share your progress.
```

* **<a href="https://github.com/join">GitHub Sign Up</a>**

> Git is a powerful version control system that aids in tracking changes in computer files and coordinating work among multiple contributors. Its basic workflow involves several key commands, including:

Reviewer: ```suggestion
GitHub relies on Git, a powerful version control system that aids in tracking changes in computer files and attributing work among multiple contributors. Its basic workflow involves several key commands, including:
```

* `git init`: Initializing a local repository is the first step in version controlling a project. This is achieved by navigating to the project directory in the terminal and using the command git init. This sets up all the necessary files and data structures, transforming the current directory into a Git repository.
* `git status`: This command displays the current state of the working directory and the staging area, highlighting which changes are yet to be committed.
* `git add`: By utilizing this command, developers can include file changes in the staging area, preparing them for the next commit.
* `git commit`: This command records the changes made to the files in the repository, creating a new commit 
that encapsulates the current state of the project.
* `git push`: With this command, developers can upload the locally saved commits to a remote repository, 
making the changes accessible to others and enabling collaboration across different locations.

> 

Reviewer: ```suggestion
If this seems like a lot, that's okay! You don't need to memorize every detail of this process, but you will want to create a workflow you're comfortable. Both Git and GitHub are industry standards that you'll use multiple times a day, every day as a software engineer.

Let's get some practice in by working through the first two lessons of the [Learn Git Branching](https://learngitbranching.js.org/?locale=en_US) tutorial together. You won't need to install anything to complete these lessons.
```

> * **<a href="https://education.github.com/git-cheat-sheet-education.pdf">GIT CHEAT SHEET</a>**

Reviewer: ```suggestion
* Everyone needs a refresher sometimes. If you're feeling stuck or can't remember which git command to use next, you can refer to this **<a href="https://education.github.com/git-cheat-sheet-education.pdf">Git cheat sheet</a>**. 
```
